### PR TITLE
Prefactor image upload warning into reusable component

### PR DESCRIPTION
**Pull request by bencodeorg**
Prefactor for limiting publishing for projects with images in labs beyond Sprite Lab. Pull out the image upload warning dialog  that appears currently when you try to upload an image to a Sprite Lab project.

No change in behavior, pure refactor.

image

image

## Testing story

Updated unit tests and tested manually that:
- the dialog appeared when trying to upload a new background and costume in a new Sprite Lab project
- dialog state cleared after opening / closing dialog without confirmation
- dialog no longer appeared on subsequent uploads after first confirmation, or after page reload
- dialog shows a single checkbox for teacher accounts
- on a published project, the dialog saying you can't upload to a published project appeared

### CODE REVIEW

**Review by breville — APPROVED** [review]
Looks good, and appreciate the detailed testing notes.  

I didn't check the migrated code line by line, but the transition from functions to functional components looks good.  

Did you consider TypeScript?  Not crucial, but it does add some extra safety.

File: apps/src/p5lab/AnimationPicker/AnimationUploadButton.jsx [hunk 1]
```diff
@@ -196,16 +104,22 @@ export function UnconnectedAnimationUploadButton({
...
-    setNoPIIConfirmed(false);
     setIsUploadModalOpen(false);
     exitedUploadWarning();
   }
 
   return (
     <>
-      {renderUploadModal()}
-      {renderPublishedWarningModal()}
+      <ImageUploadModal
```
Comment (fisher-alice) [inline apps/src/p5lab/AnimationPicker/AnimationUploadButton.jsx:113]: When I first read the name of this file, I expected that it woud only render the button. A bit out of scope of this PR, but what do you think of renaming this component, maybe to `AnimationUploader` or `ImageUploader`?

Comment (bencodeorg) [reply apps/src/p5lab/AnimationPicker/AnimationUploadButton.jsx:113]: Totally agree -- I have a couple follow-ups here, I'll try and sneak in a rename in one of those.

**Comment by bencodeorg** [comment]
> Did you consider TypeScript? Not crucial, but it does add some extra safety.

I didn't, honestly. I'm not quite over the startup hump for working in TS, so not second nature yet. Starting to get a feel for it with the Dance Party AI work though!

**Pull request merged**